### ISSUE TITLE
`PODS_TARGET_SRCROOT` does not work in `user_target_xcconfig`.

### ISSUE THREAD

**Issue by alloy** [CocoaPods/CocoaPods#6440]
Was it a deliberate decision to only make this work in `pod_target_xcconfig`?

Currently, when trying to use it in `user_target_xcconfig`, the variable is probably blank and e.g. `$(PODS_TARGET_SRCROOT)/ReactCommon` expands to `/ReactCommon`.

**Comment by alloy** [comment]
This is re the code introduced in #6037.

**Comment by segiddins** [comment]
I don't think its possible to do that, as _each_ pod will have their own value of `PODS_TARGET_SRCROOT`

**Comment by alloy** [comment]
Yeah good point, I was thinking about expanding the paths during `pod install`, but it’s definitely not how it would normally work indeed.

**Comment by segiddins** [comment]
An option would be to add `PODS_TARGET_SRCROOT_#{POD_NAME}`, but that's adding a lot of build settings

**Comment by alloy** [comment]
Would it really matter? I guess it slows down header lookup slightly, but I doubt by much. Alternatively it could be emitted only when any `user_target_xcconfig` setting refers to `PODS_TARGET_SRCROOT_#{POD_NAME}` ?

**Comment by dnkoutso** [comment]
Will see what I can do.
